mongoex/mongo.go

```diff
@@ -27,7 +28,13 @@ func New(ctx context.Context, appName string, cfg Config) (client *mongo.Client,
 	defer o11y.End(span, &err)
 
 	mongoURL, err := url.Parse(cfg.URI)
-	if err != nil {
+
+	// url.Parse will print the URI if it can't parse. The URI contains the password, so this gets the underlying error
+	// without printing the secret string.
+	var urlError *url.Error
+	if err != nil && errors.As(err, &urlError) {
```

Comment:
I feel like the err != nil is redundant (not absolutely 100%, though)

Reply:
Tested that and you're correct. 👍